### PR TITLE
Async repo take 2

### PR DESCRIPTION
**Pull request by thedaniel**
After implementing almost everything else in #9213, I ran into some
sporadic spec failures that I was having trouble reasoning about. This
is an attempt to start over with the class _beginning_ with the
functionality that went flaky, to better understand how it is tested
and whether it was being broken by an interaction with something else
in the new implementation, broken intrinsically, or had buggy tests.

This may supersede the other PR, or it may not. Let’s see what we learn.

Also, the spec is written in es6 and not coffee in this branch, and I do plan to take Nathan’s advice and update it to use async/await rather than the jasmine queue.

cc @joshaber

### CODE REVIEW

File: src/git-repository-async.js [hunk 1]
```diff
@@ -0,0 +1,202 @@
+'use babel'
+
+const path = require('path')
```
Comment (vramana) [inline src/git-repository-async.js]: @thedaniel Why not use es6 import syntax ??

**Comment by thedaniel** [comment]
Moving back over to the original PR, @joshaber found the underlying cause of the flakiness there.

**Comment by thedaniel** [comment]
False alarm, reopening

**Comment by joshaber** [comment]
@thedaniel https://github.com/atom/atom/pull/9213 is back on track, I believe. The tests pass locally and on Travis. The build's just timing on.

**Comment by controversial** [comment]
@thedaniel Why was this closed the second time?

**Comment by thedaniel** [comment]
@controversial because this was an alternate strategy for the same goal as #9213, which was ultimately merged, making this unnecessary.

**Pull request closed without merging**